test/e2e/storage/persistent_volumes-local.go

```diff
@@ -105,43 +110,286 @@ var _ = framework.KubeDescribe("[Volume] PersistentVolumes-local [Feature:LocalP
 			testVol = nil
 		})
 
-		It("should be able to mount and read from the volume", func() {
+		It("should be able to mount and read from the volume using one-command containers", func() {
 			By("Initializing test volume")
 			testVol = setupLocalVolume(config, node)
 
 			By("Creating local PVC and PV")
 			createLocalPVCPV(config, testVol)
 
-			By("Creating a pod to consume the PV")
-			readCmd := fmt.Sprintf("cat /mnt/volume1/%s", testFile)
-			podSpec := createLocalPod(config, testVol, readCmd)
-			f.TestContainerOutput("pod consumes PV", podSpec, 0, []string{testContents})
+			By("Creating a pod to read from the PV")
+			_, readCmd := createWriteAndReadCmds(volumeDir, testFile, "" /*writeTestFileContent*/)
+			podSpec := makeLocalPod(config, testVol, readCmd)
+			f.TestContainerOutput("pod reads PV", podSpec, 0, []string{testFileContent})
 		})
 
-		It("should be able to mount and write to the volume", func() {
+		It("should be able to mount and write to the volume using one-command containers", func() {
 			By("Initializing test volume")
 			testVol = setupLocalVolume(config, node)
 
 			By("Creating local PVC and PV")
 			createLocalPVCPV(config, testVol)
 
 			By("Creating a pod to write to the PV")
-			testFilePath := filepath.Join("/mnt/volume1", testFile)
-			cmd := fmt.Sprintf("echo %s > %s; cat %s", testVol.hostDir, testFilePath, testFilePath)
-			podSpec := createLocalPod(config, testVol, cmd)
+			writeCmd, _ := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
+			podSpec := makeLocalPod(config, testVol, writeCmd)
 			f.TestContainerOutput("pod writes to PV", podSpec, 0, []string{testVol.hostDir})
 		})
 	})
+
+	Context("when two pods request one prebound PVC serially", func() {
+		var (
+			testVol *localTestVolume
+			node    *v1.Node
+		)
+
+		BeforeEach(func() {
+			// Choose the first node
+			node = &config.nodes[0]
+		})
+
+		AfterEach(func() {
+			cleanupLocalVolume(config, testVol)
```

Comment:
@msau42 - I'm going to change the Logf to Failf inside cleanupLocalVolume.

framework.Logf("AfterEach: Failed to delete PV and/or PVC: %v", utilerrors.NewAggregate(errs))

to

framework.Failf("Failed to delete PV and/or PVC: %v", utilerrors.NewAggregate(errs))